### PR TITLE
[Fix] EP+FSDP checkpoint save for MoE expert parameters

### PR DESCRIPTION
**Pull request by kevssim**
# PR type
- [x] Bug Fix
- [ ] New Feature
- [ ] Document Updates
- [ ] More Models or Datasets Support

Issue: Under the EP+FSDP mode, expert parameters are sharded across both dim-0 (EP) and dim-1 (FSDP). The original `save()` call uses `DTensor.full_tensor()`, which only performs all-gather within its mesh, reconstructing only the FSDP shard while the EP dimension still retains only the local expert slice. This results in only the expert parameters from rank 0 being saved.

Fix: Added a new `_get_full_state_dict()` method to perform an additional all-gather along dim-0 within the EP group for expert parameters, reconstructing the complete expert tensor.

### CODE REVIEW

File: src/twinkle/model/transformers/transformers.py [hunk 1]
```diff
@@ -823,14 +823,21 @@ def save(self, name: Optional[str] = None, output_dir: Optional[str] = None, int
...
+            for key, value in full_sd.items():
+                processed_state_dict[key] = value
+        else:
+            # LoRA adapter save — no EP experts involved
+            state_dict = self.get_state_dict(adapter_name=adapter_name, **kwargs)
+            processed_state_dict = {}
+            save_kwargs = {}
+            for key, value in state_dict.items():
+                key = key.replace(f'.{adapter_name}.', '.')
+                processed_state_dict[key] = torch_util.to_local_tensor(value).cpu()
```
[the remark this answers is not in the document]
Comment (kevssim) [reply src/twinkle/model/transformers/transformers.py:823]: fix
